src/Template/Serialization.php

```diff
@@ -138,8 +140,10 @@ public function renderLoopArrayEmpty(string $jsonPath): string
         ]);
     }
 
-    public function renderLoopHashmap(string $jsonPath, string $propertyAccessor, string $indexVariable, string $code): string
+    public function renderLoopHashmap(string $jsonPath, string $propertyAccessor, string $indexVariable, string $code, bool $isCollection): string
     {
+        $propertyAccessor = $isCollection ? $propertyAccessor . '->toArray()' : $propertyAccessor;
```

Comment:
```suggestion
        if ($isCollection) {
            $propertyAccessor .= '->toArray()';
        }
```

Reply:
with an annotation like `array<string>|Collection<string>`, we can't be sure that the field must be a collection, it could also be an array. should we generate code that checks if `$propertyAccessor instanceof Collection` and otherwise treats the variable as native `array`?

or is that overcomplicating it and we should rather fail? depending on your usecase, you might use the 2 interchangeably - its not very clean code, but you can `foreach` without knowing which it is...

Reply:
yeah i agree that its not clean to mix the 2 - but looking at the diff, the cost seems ok to me to do this to avoid nasty surprises.